src/pages/Account/Questions.tsx

```diff
@@ -49,10 +49,6 @@ export const Questions: React.FC = () => {
 
   const { register, handleSubmit, formState } = useForm<QuestionsFormData>({
     mode: "onChange",
-    // @ts-ignore @debt Figure a way to type this properly
-    defaultValues: {
-      ...userWithId,
-    },
```

Comment:
this will probably corrupt the logic?

Reply:
From the `react-hook-form` docs (for v5)
> Note: Values defined in defaultValues will be injected into watch as defaultValue.

I cannot see usages of any fields from `WithId<User>` across the component. They only use `questions.name` fields.

Also, v7 docs explicitly state:
> defaultValues will be shallowly merged with form submission data.

But we use v5 versions and no fields from `WithId<User>` show up in `data` object if I check like this:
```
    async (data: CodeOfConductFormData) => {
      if (!user) return;

      console.log(data)
```
Which may mean that in v7 they changed the behaviour and now explicitly include that in the docs.